src/GoogleSignin.ios.js

```diff
@@ -1,157 +1,138 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
 
 import {
   View,
   NativeAppEventEmitter,
   NativeModules,
   requireNativeComponent,
-  ViewPropTypes,
-} from 'react-native';
+  ViewPropTypes
+} from 'react-native'
 
-const { RNGoogleSignin } = NativeModules;
-
-const RNGoogleSigninButton = requireNativeComponent('RNGoogleSigninButton', null);
+const { RNGoogleSignin } = NativeModules
+const RNGoogleSigninButton = requireNativeComponent('RNGoogleSigninButton', null)
 
 class GoogleSigninButton extends Component {
   static propTypes = {
     ...ViewPropTypes,
     size: PropTypes.number,
     color: PropTypes.number
-  };
-
-  componentDidMount() {
-    this._clickListener = NativeAppEventEmitter.addListener('RNGoogleSignInWillDispatch', () => {
-      GoogleSigninSingleton.signinIsInProcess = true;
-      this.props.onPress && this.props.onPress();
-    });
   }
 
-  componentWillUnmount() {
-    this._clickListener && this._clickListener.remove();
+  handleOnPress = () => {
+    if (this.props.onPress) {
+      this.props.onPress()
+      return
+    }
+
+    if (__DEV__) {
+      console.warn('RNGoogleSignin: Missing onPress prop.')
```

Comment:
I think it's more fitting to put `onPress` into `static propTypes` and not have this condition here.

Reply:
Yup, true.